src/test/java/fi/hsl/transitdata/cancellation/util/TimeUtilsTest.java

```diff
@@ -0,0 +1,124 @@
+    
+    @Test
+    public void getDatesOneDate() {
+        LocalDateTime validFrom = getTestDate("2024-01-04 10:05:00");
+        LocalDateTime validTo = getTestDate("2024-01-04 18:30:45");
+        
+        List<String> dates = TimeUtils.getDates(validFrom, validTo);
+        assertEquals(1, dates.size());
+        assertEquals("20240104", dates.get(0));
+    }
+    
+    @Test
+    public void getDatesFourDatesInTwoMonths() {
+        LocalDateTime validFrom = getTestDate("2024-01-30 10:05:00");
+        LocalDateTime validTo = getTestDate("2024-02-02 18:30:45");
+        
+        List<String> dates = TimeUtils.getDates(validFrom, validTo);
+        assertEquals(4, dates.size());
+        assertEquals("20240130", dates.get(0));
+        assertEquals("20240131", dates.get(1));
+        assertEquals("20240201", dates.get(2));
+        assertEquals("20240202", dates.get(3));
+    }
+    
+    @Test
+    public void getDatesValidFromIsNull() {
+        LocalDateTime validFrom = null;
+        LocalDateTime validTo = getTestDate("2024-01-06 18:30:45");
+        
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> TimeUtils.getDates(validFrom, validTo));
+        assertTrue(thrown.getMessage().equals("validFrom is null"));
+    }
+    
+    @Test
+    public void getDatesValidToIsNull() {
+        LocalDateTime validFrom = getTestDate("2024-01-06 18:30:45");
+        LocalDateTime validTo = null;
+        
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> TimeUtils.getDates(validFrom, validTo));
+        assertTrue(thrown.getMessage().equals("validTo is null"));
+    }
+    
+    @Test
+    public void getDatesBothInputParametersAreNull() {
+        LocalDateTime validFrom = null;
+        LocalDateTime validTo = null;
+        
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> TimeUtils.getDates(validFrom, validTo));
+        assertTrue(thrown.getMessage().equals("validFrom and validTo are null"));
+    }
+    
+    @Test
+    public void getDatesValidFromIsAfterValidTo() {
+        LocalDateTime validFrom = getTestDate("2024-01-06 18:30:45");
+        LocalDateTime validTo = getTestDate("2024-01-04 10:05:00");
+        
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> TimeUtils.getDates(validFrom, validTo));
+        assertTrue(thrown.getMessage().equals("validFrom is after validTo"));
+    }
+    
+    @Test
+    public void getDate() {
+        String dateAsString = "20240108";
+        String timeAsString = "1020";
+        
+        LocalDateTime outputDate = TimeUtils.getDate(dateAsString, timeAsString);
+        
+        assertEquals("2024-01-08 10:20:00", outputDate.format(DATETIMEFORMATTER));
+    }
+    
+    @Test
+    public void getShortDate() {
+        int unixTimestampInSeconds = 1704319200;
+        String dateAsString = TimeUtils.getShortDate(unixTimestampInSeconds);
+        assertEquals("20240104", dateAsString);
```

Comment:
This test will currently always fail if timezone isn't same as on local device. Might not need changes if proposed changes to TimeUtils.getShortDate go through.

Reply:
Since the changes proposed were stupid (My bad, didn't think through what actually the method did. Good catch!) this needs hardcoded timezone on assert to make sure it's always correct no matter what the timezone. Might be good idea either way so if implementations change further down the line the test still works.